src/fields/field_fe.cc

```diff
@@ -290,11 +311,148 @@ void FieldFE<spacedim, Value>::value_list (const std::vector< Point >  &point_li
+    }
+
+	bih_tree_ = new BIHTree( source_mesh_ );
+
+	field_name_ = rec.val<std::string>("field_name");
+
+	map1_ = new MappingP1<1,3>();
+	map2_ = new MappingP1<2,3>();
+	map3_ = new MappingP1<3,3>();
+}
+
+
+
+template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::set_mesh(const Mesh *mesh, bool boundary_domain) {
+	// Mesh can't be set for field which is not initialized.
+	if (field_name_ == "") return;
+
+	dh_ = new DOFHandlerMultiDim( const_cast<Mesh &>(*mesh) );
+	dh_->distribute_dofs(fe1, fe2, fe3);
+    unsigned int ndofs = max(dh_->fe<1>()->n_dofs(), max(dh_->fe<2>()->n_dofs(), dh_->fe<3>()->n_dofs()));
+    dof_indices = new unsigned int[ndofs];
+    // allocate data_
+	unsigned int data_size = dh_->n_global_dofs();
+	Vec v;
+    std::vector<double> data_vec;
+	data_vec.resize(data_size);
+	VecCreateSeqWithArray(PETSC_COMM_SELF, 1, data_size, &(data_vec[0]), &v);
+    data_vec_ = &v;
+    VecGetArray(*data_vec_, &data_);
+}
+
+
+
+template <int spacedim, class Value>
+bool FieldFE<spacedim, Value>::set_time(const TimeStep &time) {
+	// Time can't be set for field which is not initialized.
+	if (field_name_ == "") return false;
+
+	ASSERT_PTR(dh_)(field_name_).error("Null target mesh pointer of elementwise field, did you call set_mesh()?\n");
+    if ( reader_file_ == FilePath() ) return false;
+
+    //walkaround for the steady time governor - there is no data to be read in time==infinity
+    //TODO: is it possible to check this before calling set_time?
+    //if (time == numeric_limits< double >::infinity()) return false;
+
+    // value of last computed element must be recalculated if time is changed
+    computed_elm_idx_ = numeric_limits<unsigned int>::max();
+
+    GMSH_DataHeader search_header;
+    search_header.actual = false;
+    search_header.field_name = field_name_;
+    search_header.n_components = this->value_.n_rows() * this->value_.n_cols();
+    search_header.n_entities = source_mesh_->element.size();
+    search_header.time = time.end();
+
+    bool boundary_domain_ = false;
+    std::vector<double> data_vec = *(ReaderInstances::instance()->get_reader(reader_file_)->template get_element_data<double>(search_header,
+    		source_mesh_->elements_id_maps(boundary_domain_), this->component_idx_));
+    std::cout << std::endl;
+    std::vector<double> sum_val(4);
+    std::vector<unsigned int> elem_count(4);
+
+    FOR_ELEMENTS( dh_->mesh(), ele ) {
+    	searched_elements_.clear();
+    	((BIHTree *)bih_tree_)->find_point(ele->centre(), searched_elements_);
+    	found_point_.SetCoord( ele->centre()(0), ele->centre()(1), ele->centre()(2) );
+    	std::fill(sum_val.begin(), sum_val.end(), 0.0);
+    	std::fill(elem_count.begin(), elem_count.end(), 0);
+    	for (std::vector<unsigned int>::iterator it = searched_elements_.begin(); it!=searched_elements_.end(); it++) {
+    		ElementFullIter elm = source_mesh_->element( *it );
+			switch (elm->dim()) {
+				case 0: {
```

Comment:
We should get rid of ngh. Possibly this kind of check is already part of Element ...